### PR TITLE
Fixed #23049 -- Added %a and %A support to Date.strftime.

### PR DESCRIPTION
**Pull request by sarahboyce**
Added support for the format that was originally asked for in the ticket and then added the admonition that not all formats are supported. 

Ticket: https://code.djangoproject.com/ticket/23049

### CODE REVIEW

File: js_tests/admin/core.test.js [hunk 1]
```diff
@@ -50,11 +50,22 @@ QUnit.test('Date.getFullMonthName', function(assert) {
...
+    assert.equal(new Date(2020, 0, 26).getFullDayName(), 'Sunday', 'jan 26 2020 is a sunday');
+    assert.equal(new Date(2020, 9, 26).getFullDayName(), 'Monday', 'oct 26 2020 is a monday');
+});
+
 QUnit.test('Date.strftime', function(assert) {
     const date = new Date(2014, 6, 1, 11, 0, 5);
     assert.equal(date.strftime('%Y-%m-%d %H:%M:%S'), '2014-07-01 11:00:05');
     assert.equal(date.strftime('%B %d, %Y'), 'July 01, 2014');
     assert.equal(date.strftime('%b %d, %Y'), 'Jul 01, 2014');
+    assert.equal(date.strftime('%a %d %m %y'), 'Tue 01 07 14');
```
Comment (felixxm) [inline js_tests/admin/core.test.js:68]: Add test for `%A`.

File: js_tests/admin/core.test.js [hunk 2]
```diff
@@ -50,11 +50,23 @@ QUnit.test('Date.getFullMonthName', function(assert) {
...
+    assert.equal(new Date(2020, 9, 26).getFullDayName(), 'Monday', 'oct 26 2020 is a monday');
+});
+
 QUnit.test('Date.strftime', function(assert) {
     const date = new Date(2014, 6, 1, 11, 0, 5);
     assert.equal(date.strftime('%Y-%m-%d %H:%M:%S'), '2014-07-01 11:00:05');
     assert.equal(date.strftime('%B %d, %Y'), 'July 01, 2014');
     assert.equal(date.strftime('%b %d, %Y'), 'Jul 01, 2014');
+    assert.equal(date.strftime('%a %d %m %y (%x %X)'), 'Tue 01 07 14 (01/07/2014 11:00:05)');
+    assert.equal(date.strftime('%A (day %w of week) %I %p'), 'Tuesday (day 02 of week) 11 AM');
```
Comment (felixxm) [inline js_tests/admin/core.test.js:69]: You're trying to sneak in tests for other formats :wink:

Comment (sarahboyce) [reply js_tests/admin/core.test.js]: Tried to sneak too many in :sweat_smile: a few of these are dependant on your local timezone

File: docs/ref/forms/widgets.txt [hunk 3]
```diff
@@ -628,6 +633,11 @@ These widgets make use of the HTML elements ``input`` and ``textarea``.
     If microseconds are required, use a subclass with the
     :attr:`~Widget.supports_microseconds` attribute set to ``True``.
 
+    .. admonition:: Not all format codes are supported.
+
+        This widget does not support ``%f``, ``%j``, ``%U``, ``%W``, ``%z``,
+        ``%Z``.
```
[the remark this answers is not in the document]
Comment (sarahboyce) [reply docs/ref/forms/widgets.txt:639]: `%U` `%W` and `%j` are not handled in the JavaScript function `Date.prototype.strftime`, they can be added in though (just thought they were not wanted enough compared to the code required to calculate them).

You're right that we don't suggest that we support all formats, but I wasn't sure if it's useful to state that the admin widget is more restrictive then perhaps people think? (I can remove` %f` `%z` and` %Z`)

[the remark this answers is not in the document]
Comment (felixxm) [reply docs/ref/forms/widgets.txt:639]: > You're right that we don't suggest that we support all formats, but I wasn't sure if it's useful to state that the admin widget is more restrictive then perhaps people think? (I can remove` %f` `%z` and` %Z`)

Yes, I'd remove them to don't suggest that widget itself support microseconds of timezones.

[the remark this answers is not in the document]
Comment (felixxm) [reply docs/ref/forms/widgets.txt:639]: I'm also not sure if it deserves for a separate admonition, maybe:
```diff
diff --git a/docs/ref/forms/widgets.txt b/docs/ref/forms/widgets.txt
index 1fbd5ad4a0..f67004bf69 100644
--- a/docs/ref/forms/widgets.txt
+++ b/docs/ref/forms/widgets.txt
@@ -622,7 +622,8 @@ These widgets make use of the HTML elements ``input`` and ``textarea``.
 
     If no ``format`` argument is provided, the default format is the first
     format found in :setting:`DATETIME_INPUT_FORMATS` and respects
-    :doc:`/topics/i18n/formatting`.
+    :doc:`/topics/i18n/formatting`. ``%U``, ``%W``, and ``%j`` formats are not
+    supported by this widget.
 
     By default, the microseconds part of the time value is always set to ``0``.
     If microseconds are required, use a subclass with the

```

[the remark this answers is not in the document]
Comment (sarahboyce) [reply docs/ref/forms/widgets.txt]: Yeah I think that's better thank you :star:. Squashed it into a single commit as I don't think it deserves it's own one anymore :+1:

**Comment by felixxm** [comment]
@sarahboyce Thanks :+1:

**Pull request merged**